### PR TITLE
feat(settings): moved all creator settings into ProjectSettings

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description
Got my eyes onto a nice neat feature thingy, which is extending the project settings. Took a look into it because I had some time on Friday and we could also use it later on for logging in.

image

### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

**Test Configuration**:
Also needs two additional PRs:
* https://github.com/Innoactive/TextToSpeech-Component/pull/35
* https://github.com/Innoactive/XR-Interaction-Component/pull/70

Copy and paste from Assets folder for a fresh clone:

```
git clone --single-branch --branch develop git@github.com:Innoactive/Basic-Conditions-And-Behaviors.git Innoactive/Creator/Components/Basic-ConditionsBehaviors
git clone --single-branch --branch develop git@github.com:Innoactive/Basic-Interaction-Component.git Innoactive/Creator/Components/Basic-Interaction
git clone --single-branch --branch feature/pretty-settings git@github.com:Innoactive/TextToSpeech-Component.git Innoactive/Creator/Components/TextToSpeech
git clone --single-branch --branch feature/pretty-settings git@github.com:Innoactive/XR-Interaction-Component.git Innoactive/Creator/Components/XR-Interaction-Component
git clone --single-branch --branch feature/pretty-settings git@github.com:Innoactive/Creator.git Innoactive/Creator/Core
git clone --single-branch --branch develop git@github.com:Innoactive/IA-Training-Template.git Innoactive/Creator/IA-Training-Template
```

### CODE REVIEW

File: Editor/UI/CreatorMenu/ShowSettings.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+using Innoactive.CreatorEditor.Configuration;
+using UnityEditor;
+
+namespace Innoactive.CreatorEditor.CreatorMenu
+{
+    internal static class ShowSettings
```
Comment (Gusinuhe) [inline Editor/UI/CreatorMenu/ShowSettings.cs:6]: Maybe something more comprehensive like: `CreatorProjectSettings` 🤔

File: Editor/UI/ProjectSettings/AnalyticsSettingsProvider.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    const string Path = "Project/Creator/Analytics";
+
+    private CreatorProjectSettings data;
+
+    public AnalyticsSettingsProvider() : base(Path, SettingsScope.Project) {}
+
+    public static bool IsSettingsAvailable()
+    {
+        return true;
+    }
```
Comment (Gusinuhe) [inline Editor/UI/ProjectSettings/AnalyticsSettingsProvider.cs:18]: Why is this necessary? 🤔

File: Editor/UI/ProjectSettings/AnalyticsSettingsProvider.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+using Innoactive.CreatorEditor.Analytics;
+using Innoactive.CreatorEditor.UI;
+using UnityEditor;
+using UnityEngine;
+
+internal class AnalyticsSettingsProvider : SettingsProvider
+{
+    const string Path = "Project/Creator/Analytics";
+
+    private CreatorProjectSettings data;
```
Comment (Gusinuhe) [inline Editor/UI/ProjectSettings/AnalyticsSettingsProvider.cs:11]: Why is this necessary? 🤔

File: Editor/UI/ProjectSettings/CreatorPageProvider.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    [SettingsProvider]
+    public static SettingsProvider GetCreatorSettingsProvider()
+    {
+        if (IsSettingsAvailable())
+        {
+            SettingsProvider provider = new CreatorPageProvider();
+
+            // Automatically extract all keywords from the Styles.
+            //provider.keywords = GetSearchKeywordsFromGUIContentProperties<Styles>();
```
Comment (Gusinuhe) [inline Editor/UI/ProjectSettings/CreatorPageProvider.cs:47]: 🤔

File: Editor/UI/ProjectSettings/CreatorSettingProvider.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
+﻿using Innoactive.Creator.Core.Utils.Logging;
+using Innoactive.CreatorEditor.UI;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+internal class CreatorSettingProvider : SettingsProvider
+{
+    const string Path = "Project/Creator/Settings";
```
Comment (Maxsiom) [inline Editor/UI/ProjectSettings/CreatorSettingProvider.cs:9]:
```suggestion
    const string Path = "Project/Creator/Logging";
```

Would call it Logging. `Settings` sounds too generic :)

File: .Documentation/articles/innoactive-creator/snap-zones.md [hunk 6]
```diff
@@ -50,7 +50,7 @@ To automatically create a matching snap zone, click the `Create Snap Zone` butto
 
 ### III. Modify Default Snap Zone Settings (optional)
 
-You can either change the settings directly in the new snap zone object or change the default settings in the `Snap Zone Settings` window. In order to open the default snap zone settings, **go to the Unity's toolbar** and select: `Innoactive` > `Settings` > `Snap Zone Settings`. All changes are saved automatically, and every snap zone that will be created via the `Create Snap Zone` button afterwards will get the new settings by default.
+You can either change the settings directly in the new snap zone object or change the default settings in the `Snap Zone Settings`. In order to open the default snap zone settings, **go to the Unity's toolbar** and select: `Innoactive` > `Settings` to open the Creator part of the ProjectSettings, there you will find an entry for the `Snap Zone` to configure the default values. All changes are saved automatically, and every snap zone that will be created via the `Create Snap Zone` button afterwards will get the new settings by default.
```
Comment (Maxsiom) [inline .Documentation/articles/innoactive-creator/snap-zones.md:53]:
```suggestion
You can either change the settings directly in the new snap zone object or change the default settings in the `Snap Zone Settings`. In order to open the default snap zone settings, **go to the Unity's toolbar** and select: `Innoactive` > `Settings` to open the Creator part of the `ProjectSettings`. There you will find an entry for the `Snap Zone` to configure the default values. All changes are saved automatically, and every snap zone that will be created via the `Create Snap Zone` button afterwards will get the new settings by default.
```

File: .Documentation/articles/developer/12-text-to-speech.md [hunk 7]
```diff
@@ -13,7 +13,7 @@ Supported text to speech engines:
 There is also an option to integrate additional text to speech engines.
 
 ## How to Setup a Text to Speech Engine (TTS)
-The configuration is stored in a scriptable object called TextToSpeechConfiguration which is accessible via the Innoactive menu `Innoactive > Settings > TextToSpeech Settings`. It is usable out of the box and configured by default to use the Windows text to speech API which allows to create audio files on any Windows 10 machine without internet connection.
+The configuration is stored in a scriptable object called TextToSpeechConfiguration which is accessible via the ProjectSettings or the Innoactive menu `Innoactive > Settings` (which will open the ProjectSettings). Here you can find the Text to speech entry under `Creator > Text to Speech`. It is usable out of the box and configured by default to use the Windows text to speech API which allows to create audio files on any Windows 10 machine without internet connection.
```
Comment (Maxsiom) [inline .Documentation/articles/developer/12-text-to-speech.md:16]:
```suggestion
The configuration is stored in a scriptable object called TextToSpeechConfiguration which is accessible via the ProjectSettings or the Innoactive menu `Innoactive > Settings` (which will open the `ProjectSettings`). Here you can find the Text to speech entry under `Creator > Text to Speech`. It is usable out of the box and configured by default to use the Windows text to speech API which allows to create audio files on any Windows 10 machine without internet connection.
```

File: Editor/UI/Wizard/Setup/AnalyticsPage.cs [hunk 8]
```diff
@@ -32,7 +32,7 @@ public override void Draw(Rect window)
...
 
                 GUILayout.Label("We Are Transparent", CreatorEditorStyles.Header);
-                GUILayout.Box("The Innoactive Creator is open-source. Feel free to check our analytics code in <b>Core/Editor/Analytics</b>\n\nIf you want to opt-out of tracking, open <b>Innoactive > Settings > Analytics Settings</b> in the Unity's menu bar and choose <i>disabled</i> from the drop-down menu.", CreatorEditorStyles.Paragraph);
+                GUILayout.Box("The Innoactive Creator is open-source. Feel free to check our analytics code in <b>Core/Editor/Analytics</b>\n\nIf you want to opt-out of tracking, open the ProjectSettings under <b>Creator > Analytics</b> and choose <i>disabled</i> from the drop-down menu.", CreatorEditorStyles.Paragraph);
```
Comment (Maxsiom) [inline Editor/UI/Wizard/Setup/AnalyticsPage.cs:35]:
```suggestion
                GUILayout.Box("The Innoactive Creator is open-source. Feel free to check our analytics code in <b>Core/Editor/Analytics</b>\n\nIf you want to opt-out of tracking, open the 'ProjectSettings' by navigating to <b>Innoactive > Settings</b>, find <b>Creator > Analytics</b>, and choose <i>disabled</i> from the drop-down menu.", CreatorEditorStyles.Paragraph);
```

**Review by Maxsiom — CHANGES_REQUESTED** [review]
Very nice! Looks great.

**Review by Gusinuhe — COMMENTED** [review]
Can we somehow recycle the GUI content from the wizard pages?
Can we keep the buttons in the `Innoactive` menu from the toolbar?

File: Editor/UI/ProjectSettings/CreatorSettingProvider.cs (hunk 5 above)
Comment (SimonTheSourcerer) [reply Editor/UI/ProjectSettings/CreatorSettingProvider.cs:9]: thought about adding more stuff into the settings page later on.

**Review by Maxsiom — APPROVED** [review]

File: Editor/UI/ProjectSettings/CreatorPageProvider.cs [hunk 9]
```diff
@@ -0,0 +1,40 @@
+﻿using Innoactive.CreatorEditor.UI;
+using UnityEditor;
+using UnityEngine;
+
+internal class CreatorPageProvider : SettingsProvider
+{
+    const string Path = "Project/Creator";
+
+    private CreatorProjectSettings data;
```
Comment (Gusinuhe) [inline Editor/UI/ProjectSettings/CreatorPageProvider.cs:9]: This is never used

Comment (SimonTheSourcerer) [inline Editor/UI/ProjectSettings/CreatorPageProvider.cs:9]: ``
```suggestion

```

**Review by Gusinuhe — APPROVED** [review]

**Review by Maxsiom — APPROVED** [review]
